clients/da-vinci-client/src/main/java/com/linkedin/davinci/client/AvroGenericDaVinciClient.java

```diff
@@ -677,13 +678,14 @@ private void initBackend(
       VeniceConfigLoader configLoader,
       Optional<Set<String>> managedClients,
       ICProvider icProvider,
-      Optional<ObjectCacheConfig> cacheConfig) {
+      Optional<ObjectCacheConfig> cacheConfig,
+      DaVinciRecordTransformer recordTransformer) {
     synchronized (AvroGenericDaVinciClient.class) {
       if (daVinciBackend == null) {
         logger
             .info("Da Vinci Backend does not exist, creating a new backend for client: " + clientConfig.getStoreName());
         daVinciBackend = new ReferenceCounted<>(
-            new DaVinciBackend(clientConfig, configLoader, managedClients, icProvider, cacheConfig),
+            new DaVinciBackend(clientConfig, configLoader, managedClients, icProvider, cacheConfig, recordTransformer),
```

Comment:
This will always pass null, you need to bring the line at 735 up before this, or just pass in the value of daVinciConfig.getRecordTransformer (and maybe forgo the member variable entirely from this class?